### ISSUE TITLE
Removing websites or stores together with their configuration from config.php fails

### ISSUE THREAD

**Issue by schmengler** [magento/magento2#24061]
### Preconditions (*)
* Magento 2.3.2 (and probably all earlier versions)
* websites and stores created via `scopes`  section in `app/etc/config.php`
* any configuration on website level (e.g. base urls) in the `system` section in `app/etc/config.php`

### Steps to reproduce (*)

1. remove a website together with its scope specific configuration from `app/etc/config.php`
2. run `bin/magento app:config:import`

### Expected result (*)

1. The website is removed
2. Other configuration updates are applied
3. The shop continues to work

### Actual result (*)

1. The website is removed
2. The command aborts with error message: "The website with code xxx that was requested wasn't found. Verify the website and try again."
3. The shop throws an error: "Config files have changed. Run app:config:import or setup:upgrade command to synchronize configuration."

Stack trace from exception log:
```
         "trace" : [
            "/var/www/html/src/vendor/magento/module-deploy/Console/Command/App/ConfigImport/Processor.php:146",
            "/var/www/html/src/vendor/magento/module-deploy/Console/Command/App/ConfigImportCommand.php:63",
            "/var/www/html/src/vendor/symfony/console/Command/Command.php:255",
            "/var/www/html/src/vendor/magento/framework/Interception/Interceptor.php:58",
            "/var/www/html/src/vendor/magento/framework/Interception/Interceptor.php:138",
            "/var/www/html/src/vendor/magento/framework/Interception/Interceptor.php:153",
            "/var/www/html/src/generated/code/Magento/Deploy/Console/Command/App/ConfigImportCommand/Interceptor.php:104",
            "/var/www/html/src/vendor/symfony/console/Application.php:893",
            "/var/www/html/src/vendor/symfony/console/Application.php:262",
            "/var/www/html/src/vendor/magento/framework/Console/Cli.php:105",
            "/var/www/html/src/vendor/symfony/console/Application.php:145",
            "/var/www/html/src/bin/magento:23"
         ],
         "class" : "Magento\\Framework\\Exception\\State\\InvalidTransitionException",
         "previous" : {
            "class" : "Magento\\Framework\\Exception\\RuntimeException",
            "code" : 0,
            "previous" : {
               "class" : "Magento\\Framework\\Exception\\NoSuchEntityException",
               "code" : 0,
               "file" : "/var/www/html/src/vendor/magento/module-store/Model/WebsiteRepository.php:81",
               "message" : "The website with code eu that was requested wasn't found. Verify the website and try again.",
               "trace" : [
                  "/var/www/html/src/generated/code/Magento/Store/Model/WebsiteRepository/Interceptor.php:24",
                  "/var/www/html/src/vendor/magento/module-store/Model/StoreManager.php:209",
                  "/var/www/html/src/generated/code/Magento/Store/Model/StoreManager/Interceptor.php:102",
                  "/var/www/html/src/generated/code/Magento/Store/Model/StoreManagerInterface/Proxy.php:135",
                  "/var/www/html/src/vendor/magento/module-store/Model/Resolver/Website.php:31",
                  "/var/www/html/src/vendor/magento/module-config/Model/PreparedValueFactory.php:125",
                  "/var/www/html/src/vendor/magento/module-config/Model/Config/Importer/SaveProcessor.php:101",
                  "/var/www/html/src/vendor/magento/module-config/Model/Config/Importer/SaveProcessor.php:67",
                  "/var/www/html/src/vendor/magento/module-config/Model/Config/Importer.php:131",
                  "{\"function\":\"Magento\\\\Config\\\\Model\\\\Config\\\\{closure}\",\"class\":\"Magento\\\\Config\\\\Model\\\\Config\\\\Importer\",\"type\":\"->\",\"args\":[]}",
                  "/var/www/html/src/vendor/magento/framework/App/State.php:187",
                  "/var/www/html/src/generated/code/Magento/Framework/App/State/Interceptor.php:89",
                  "/var/www/html/src/vendor/magento/module-config/Model/Config/Importer.php:132",
                  "/var/www/html/src/vendor/magento/module-deploy/Console/Command/App/ConfigImport/Processor.php:146",
                  "/var/www/html/src/vendor/magento/module-deploy/Console/Command/App/ConfigImportCommand.php:63",
                  "/var/www/html/src/vendor/symfony/console/Command/Command.php:255",
                  "/var/www/html/src/vendor/magento/framework/Interception/Interceptor.php:58",
                  "/var/www/html/src/vendor/magento/framework/Interception/Interceptor.php:138",
                  "/var/www/html/src/vendor/magento/framework/Interception/Interceptor.php:153",
                  "/var/www/html/src/generated/code/Magento/Deploy/Console/Command/App/ConfigImportCommand/Interceptor.php:104",
                  "/var/www/html/src/vendor/symfony/console/Application.php:893",
                  "/var/www/html/src/vendor/symfony/console/Application.php:262",
                  "/var/www/html/src/vendor/magento/framework/Console/Cli.php:105",
                  "/var/www/html/src/vendor/symfony/console/Application.php:145",
                  "/var/www/html/src/bin/magento:23"
               ]
            },
            "file" : "/var/www/html/src/vendor/magento/module-config/Model/PreparedValueFactory.php:146",
            "trace" : [
               "/var/www/html/src/vendor/magento/module-config/Model/Config/Importer/SaveProcessor.php:101",
               "/var/www/html/src/vendor/magento/module-config/Model/Config/Importer/SaveProcessor.php:67",
               "/var/www/html/src/vendor/magento/module-config/Model/Config/Importer.php:131",
               "{\"function\":\"Magento\\\\Config\\\\Model\\\\Config\\\\{closure}\",\"class\":\"Magento\\\\Config\\\\Model\\\\Config\\\\Importer\",\"type\":\"->\",\"args\":[]}",
               "/var/www/html/src/vendor/magento/framework/App/State.php:187",
               "/var/www/html/src/generated/code/Magento/Framework/App/State/Interceptor.php:89",
               "/var/www/html/src/vendor/magento/module-config/Model/Config/Importer.php:132",
               "/var/www/html/src/vendor/magento/module-deploy/Console/Command/App/ConfigImport/Processor.php:146",
               "/var/www/html/src/vendor/magento/module-deploy/Console/Command/App/ConfigImportCommand.php:63",
               "/var/www/html/src/vendor/symfony/console/Command/Command.php:255",
               "/var/www/html/src/vendor/magento/framework/Interception/Interceptor.php:58",
               "/var/www/html/src/vendor/magento/framework/Interception/Interceptor.php:138",
               "/var/www/html/src/vendor/magento/framework/Interception/Interceptor.php:153",
               "/var/www/html/src/generated/code/Magento/Deploy/Console/Command/App/ConfigImportCommand/Interceptor.php:104",
               "/var/www/html/src/vendor/symfony/console/Application.php:893",
               "/var/www/html/src/vendor/symfony/console/Application.php:262",
               "/var/www/html/src/vendor/magento/framework/Console/Cli.php:105",
               "/var/www/html/src/vendor/symfony/console/Application.php:145",
               "/var/www/html/src/bin/magento:23"
            ],
            "message" : "The website with code eu that was requested wasn't found. Verify the website and try again."
         },
         "code" : 0
```

The issue seems to be that Magento wants to remove configuration entries for websites after the websites have been removed and fails at that.

To work around it, I currently have to make two separate deployments, one where I remove the website specific configuration and a second where I remove the website.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-19037` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @schmengler. 

Thank you for your report and collaboration!

The issue was fixed by Magento team. The fix was delivered into `magento/magento2:2.4-develop` branch(es).
Related commit(s):
 - [581ac89489128c3a0bcc4f8558413fbe8a4a43f8](https://github.com/magento/magento2/commit/581ac89489128c3a0bcc4f8558413fbe8a4a43f8)

The fix will be available with the upcoming `2.4.0` release.

**Comment by pushpendra07** [comment]
@magento-engcom-team Same error showing while we run index management, how can we solve this
